### PR TITLE
Format: expand pretty printer

### PR DESCRIPTION
**Pull request by d-xo**
## Description

Expand pretty printer to cover all cases. This currently produces output that is very newline heavy. Probably there is a smart way to teach it to print shorter expressions on one line only, but I think this is still a lot better than what we currently have.

## Checklist

- [x] tested locally
- [ ] added automated tests
- [ ] updated the docs
- [ ] updated the changelog

### CODE REVIEW

File: src/EVM/Format.hs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DataKinds #-}
```
Comment (msooseth) [inline src/EVM/Format.hs:2]:
```suggestion
```

File: src/EVM/Format.hs [hunk 2]
```diff
@@ -499,12 +511,134 @@ formatExpr = go
...
+      Div a b -> fmt "Div" [a, b]
+      SDiv a b -> fmt "SDiv" [a, b]
+      Mod a b -> fmt "Mod" [a, b]
+      SMod a b -> fmt "SMod" [a, b]
+      AddMod a b c -> fmt "AddMod" [a, b, c]
+      MulMod a b c -> fmt "MulMod" [a, b, c]
+      Exp a b -> fmt "Exp" [a, b]
+      SEx a b -> fmt "SEx" [a, b]
+      Min a b -> fmt "Min" [a, b]
+      Max a b -> fmt "Min" [a, b]
```
Comment (msooseth) [inline src/EVM/Format.hs:537]:
```suggestion
      Max a b -> fmt "Max" [a, b]
```

File: src/EVM/Format.hs [hunk 3]
```diff
@@ -581,16 +719,63 @@ formatExpr = go
...
+          , ")"
+          ]
+
+formatProp :: Prop -> Text
+formatProp x = T.stripEnd $ case x of
+  PEq a b -> fmt "PEq" [a, b]
+  PLT a b -> fmt "PLT" [a, b]
+  PGT a b -> fmt "PGT" [a, b]
+  PGEq a b -> fmt "PLT" [a, b]
+  PLEq a b -> fmt "PGT" [a, b]
```
Comment (msooseth) [inline src/EVM/Format.hs:740]:
```suggestion
  PGEq a b -> fmt "PGEq" [a, b]
  PLEq a b -> fmt "PLEq" [a, b]
```

**Review by msooseth — CHANGES_REQUESTED** [review]
Really cool! I think there may be a few minor typos? I'm not sure of some were intentional, though :)

**Comment by d-xo** [comment]
Thanks for spotting the typos! None were intentional 😬😅

**Review by msooseth — APPROVED** [review]
lgtm

**Pull request merged**